packages/openstef-core/src/openstef_core/feature_engineering/forecasting_transforms/trend_features.py

```diff
@@ -0,0 +1,146 @@
+# SPDX-FileCopyrightText: 2025 Contributors to the OpenSTEF project <short.term.energy.forecasts@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""Transform for extracting trend features from time series data.
+
+This module provides functionality to compute trend-based features that capture
+long-term patterns and movements in time series data, helping improve forecasting
+accuracy by identifying underlying trends.
+"""
+
+from datetime import timedelta
+from enum import StrEnum
+
+import pandas as pd
+from pydantic import Field
+
+from openstef_core.base_model import BaseConfig
+from openstef_core.datasets import TimeSeriesDataset
+from openstef_core.datasets.transforms import TimeSeriesTransform
+from openstef_core.utils import timedelta_to_isoformat
+
+
+class AggregationFunction(StrEnum):
+    """Enum of supported aggregation functions for rolling aggregates."""
+
+    MEAN = "mean"
+    MEDIAN = "median"
+    MAX = "max"
+    MIN = "min"
+
+
+class RollingAggregateFeaturesConfig(BaseConfig):
+    """Configuration for the RollingAggregateFeatures transform."""
+
+    rolling_window_size: timedelta = Field(
+        default=timedelta(hours=24),
+        description="Rolling window size for the aggregation.",
+    )
+    aggregation_functions: list[AggregationFunction] = Field(
+        default_factory=lambda: [AggregationFunction.MEDIAN, AggregationFunction.MIN, AggregationFunction.MAX],
+        description="List of aggregation functions to compute over the rolling window. ",
+    )
+
+
+class RollingAggregateFeatures(TimeSeriesTransform):
```

Comment:
The file of this feature should probably be called something like `rolling_aggregate_transform.py`? The trend feature is usually created by extrapolating with a simpler model.